src/compiler/crystal/codegen/unions.cr

```diff
@@ -0,0 +1,122 @@
+# Here lies the logic of the representation of the MixedUnionType.
+#
+# Which structure is used to represent them is defined in `LLVMTyper#create_llvm_type`.
+#
+# The `#union_type_and_value_pointer` will allow to read the current value of the union.
+# The `#store*_in_union` operations allow to write the value in a unions.
+# The `#{assign|downcast|upcast}_distinct_union_types` operation matches the
+# semantics described in `./casts.cr`
+#
+# Together these operations should encapsulate the binary representation of the MixedUnionType.
+#
+# Other unions like ReferenceUnionType that have a more trivial
+# representation are not handled here.
+#
+module Crystal
+  class LLVMTyper
+    private def create_llvm_type(type : MixedUnionType, wants_size)
+      llvm_name = llvm_name(type, wants_size)
+      if s = @structs[llvm_name]?
```

Comment:
Please, use full names.

Reply:
The scope of `s` is only the `if`, which consist of a 1 line. Otherwise, I would agree with using the longer variable name.

Reply:
Well, yeah I know, but still... it won't hurt to avoid cryptic one-letter variable names...

Reply:
I think the issue here is that a good name is probably `struct`, but that's a keyword... so one could use `a_struct` or similar. But given that it's just one line, I think `s` is fine.